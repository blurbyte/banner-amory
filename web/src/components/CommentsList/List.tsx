import styled from 'styled-components';

const List = styled.ul`
  margin: 0;
  margin-top: 1rem;
  padding: 0;
  list-style: 0;
`;

export default List;
