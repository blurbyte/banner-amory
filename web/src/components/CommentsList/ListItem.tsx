import * as React from 'react';
import styled from 'styled-components';

const Wrapper = styled.li`
  display: grid;
  grid-template-columns: 15rem 1fr;
  height: 3rem;
  align-items: end;
`;

const Name = styled.span`
  font-weight: 600;
  padding-right: 1rem;
`;

const Message = styled.span`
  font-weight: 400;
`;

type ListItemProps = {
  userName: string;
  message: string;
};

class ListItem extends React.Component<ListItemProps> {
  render() {
    const { userName, message } = this.props;

    return (
      <Wrapper>
        <Name>{userName}</Name>
        <Message>{message}</Message>
      </Wrapper>
    );
  }
}

export default ListItem;
