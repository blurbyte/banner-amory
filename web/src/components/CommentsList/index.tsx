import * as React from 'react';

import List from './List';
import ListItem from './ListItem';

class CommentsList extends React.Component {
  render() {
    return (
      <List>
        <ListItem
          userName="legend27"
          message="This item is just amazing, probably one of the best!"
        />
        <ListItem
          userName="ArianaGrande"
          message="Dunno what you talking about, looks legit 2 me"
        />
        <ListItem userName="taylorswift" message="Donâ€™t understand how is it tier1 #trash" />
      </List>
    );
  }
}

export default CommentsList;
