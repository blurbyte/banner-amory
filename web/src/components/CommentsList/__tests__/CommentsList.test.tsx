import * as React from 'react';
import * as renderer from 'react-test-renderer';

import CommentsList from '../index';

test('renders correctly', () => {
  const component = renderer.create(<CommentsList />);
  expect(component.toJSON()).toMatchSnapshot();
});
