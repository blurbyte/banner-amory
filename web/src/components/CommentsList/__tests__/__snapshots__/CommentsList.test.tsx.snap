// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ul
  className="sc-bdVaJa kRDQbv"
>
  <li
    className="sc-bwzfXH hkazQv"
  >
    <span
      className="sc-htpNat gAMvjR"
    >
      legend27
    </span>
    <span
      className="sc-bxivhb ijKRfN"
    >
      This item is just amazing, probably one of the best!
    </span>
  </li>
  <li
    className="sc-bwzfXH hkazQv"
  >
    <span
      className="sc-htpNat gAMvjR"
    >
      ArianaGrande
    </span>
    <span
      className="sc-bxivhb ijKRfN"
    >
      Dunno what you talking about, looks legit 2 me
    </span>
  </li>
  <li
    className="sc-bwzfXH hkazQv"
  >
    <span
      className="sc-htpNat gAMvjR"
    >
      taylorswift
    </span>
    <span
      className="sc-bxivhb ijKRfN"
    >
      Donâ€™t understand how is it tier1 #trash
    </span>
  </li>
</ul>
`;
